### PR TITLE
Apply inserts on scatter graph after drawing error bars

### PR DESCRIPTION
**Pull request by lshrinivas**
Otherwise inserts on error bars have stale data

### CODE REVIEW

File: lib/views/ScatterGraph.js [hunk 1]
```diff
@@ -80,9 +78,10 @@ class ScatterGraph extends Graph {
                 .attr('y2', d => this.yRange(d.y_min));
 
             errorBars.exit().remove();
-
         }
 
+        this.d3Svg.call(this._applyInserts.bind(this));
+
```
Comment (lshrinivas) [inline lib/views/ScatterGraph.js:84]: Needed to move this below the draw error bars code

File: test/views/ScatterGraph.spec.js [hunk 2]
```diff
@@ -216,7 +232,24 @@ function (
...
+
+            var newData = [
+                {x_value: 0,   y_value: 10, y_min: 5, y_max: 15},
+                {x_value: 100, y_value: 20, y_min: 15, y_max: 25},
+                {x_value: 200, y_value: 30, y_min: 25, y_max: 35}
+            ];
+            dataSeries.setData(newData);
 
+            expect(circleDataChecker.calls.allArgs()).toEqual([[newData[0]], [newData[1]], [newData[2]]]);
+            expect(errorBarDataChecker.calls.allArgs()).toEqual([[newData[0]], [newData[1]], [newData[2]]]);
```
Comment (lshrinivas) [inline test/views/ScatterGraph.spec.js:252]: Added test to ensure that both selections have up to date data in the inserts

Comment (choochburger) [reply test/views/ScatterGraph.spec.js:252]: Nice 👍

**Review by choochburger — APPROVED** [review]
Looks good 👍

**Pull request merged**